macros/identifiers.sql

```diff
@@ -0,0 +1,249 @@
+      {{ return([
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_media_session_1', 'field': 'mediaSessionId', 'prefix': 'media_session_'},
+        {'schema': 'contexts_com_snowplowanalytics_mobile_screen_1', 'field': 'id', 'prefix': 'mobile_screen_'},
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1', 'field': 'id', 'prefix': 'web_page_'}
+      ]) }}
+
+    {% elif var('snowplow__enable_web_events') and var('snowplow__enable_mobile_events') %}
+      {{ return([
+        {'schema': 'contexts_com_snowplowanalytics_mobile_screen_1', 'field': 'id', 'prefix': 'mobile_screen_'},
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1', 'field': 'id', 'prefix': 'web_page_'}
+      ]) }}
+
+    {% elif var('snowplow__enable_mobile_events') %}
+      {{ return([{'schema': 'contexts_com_snowplowanalytics_mobile_screen_1', 'field': 'id', 'prefix': 'mobile_screen_'}]) }}
+
+    {% else %}
+      {{ return([{'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1', 'field': 'id', 'prefix': 'web_page_'}] )}}
+
+    {% endif %}
+  {% endif %}
+
+{% endmacro %}
+
+{% macro bigquery__session_identifiers() %}
+
+  {% if var('snowplow__session_identifiers') %}
+    {{ return(var('snowplow__session_identifiers')) }}
+
+  {% else %}
+
+    {% if var('snowplow__enable_media_session') %}
+      {{ return([
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_media_session_1_*', 'field': 'media_session_id', 'prefix': 'media_session_'},
+        {'schema': 'contexts_com_snowplowanalytics_mobile_screen_1_*', 'field': 'id', 'prefix': 'mobile_screen_'},
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1_*', 'field': 'id', 'prefix': 'web_page_'}
+      ]) }}
+
+    {% elif var('snowplow__enable_web_events') and var('snowplow__enable_mobile_events') %}
+      {{ return([
+        {'schema': 'contexts_com_snowplowanalytics_mobile_screen_1_*', 'field': 'id', 'prefix': 'mobile_screen_'},
+        {'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1_*', 'field': 'id', 'prefix': 'web_page_'}
+      ]) }}
+
+    {% elif var('snowplow__enable_mobile_events') %}
+      {{ return([{'schema': 'contexts_com_snowplowanalytics_mobile_screen_1_*', 'field': 'id', 'prefix': 'mobile_screen_'}]) }}
+
+    {% else %}
+      {{ return([{'schema': 'contexts_com_snowplowanalytics_snowplow_web_page_1_*', 'field': 'id', 'prefix': 'web_page_'}] )}}
+
+    {% endif %}
+  {% endif %}
+
+{% endmacro %}
+
+{% macro postgres__session_identifiers() %}
+
+  {% if var('snowplow__session_identifiers') %}
+    {{ return(var('snowplow__session_identifiers')) }}
+
+  {% else %}
+
+    {% if var('snowplow__enable_media_session') %}
+      {{ return([
+        {'schema': 'com_snowplowanalytics_snowplow_media_session_1', 'field': 'media_session_id', 'prefix': 'media_session_'},
+        {'schema': 'com_snowplowanalytics_mobile_screen_1', 'field': 'id', 'prefix': 'mobile_screen_'},
```

Comment:
Do we need to check for `snowplow__enable_web_events` and `snowplow__enable_mobile_events` in these fallbacks?

Reply:
Yes we do, good spot!